include/utils/Simplify.h

```diff
@@ -160,32 +182,41 @@ class Simplify
         };
         std::priority_queue<std::pair<size_t, coord_t>, std::vector<std::pair<size_t, coord_t>>, decltype(comparator)> by_importance(comparator);
 
-        //Add the initial points.
-        for(size_t i = 0; i < polygon.size(); ++i)
-        {
-            const coord_t vertex_importance = importance(polygon, to_delete, i, is_closed);
-            by_importance.emplace(i, vertex_importance);
-        }
-
-        //Iteratively remove the least important point until a threshold.
         Polygonal result = polygon; //Make a copy so that we can also shift vertices.
-        coord_t vertex_importance = 0;
-        while(by_importance.size() > min_size)
+        for (int64_t current_removed = -1; (polygon.size() - current_removed) > min_size && current_removed != 0;)
         {
-            std::pair<size_t, coord_t> vertex = by_importance.top();
-            by_importance.pop();
-            //The importance may have changed since this vertex was inserted. Re-compute it now.
-            //If it doesn't change, it's safe to process.
-            vertex_importance = importance(result, to_delete, vertex.first, is_closed);
-            if(vertex_importance != vertex.second)
+            current_removed = 0;
```

Comment:
We need to reset the `by_importance` queue as we are re-iterating over the same data structure.
```suggestion
            current_removed = 0;
            by_importance.clear();
```

Reply:
We don't really need to, since the thing is either empty or has few enough vertices that it'll stop immediately. It can't hurt though!